surveys/static/js/charts.js

```diff
@@ -0,0 +1,336 @@
+// Make sure JQuery is loaded
+window.onload = function() {
+  if (!window.jQuery) {
+    throw new Error('JQuery must be loaded before charts.js');
+  }
+}
+
+var ChartHelper = function(surveys, types, bins) {
+  /**
+   * Helper class for displaying HighCharts charts on a page.
+   * This class depends on a number of nested objects as arguments in order to
+   * sensibly initialize chart display. For an example of the structure of these
+   * objects, see surveys.views.SurveySubmittedDetail.
+   * @constructor
+   * @param {Object} surveys - A data object containing survey results.
+   * @param {Object} types - An object of type definitions for survey results.
+   * @param {Object} bins - An object of bin definitions for binning survey results.
+   */
+  this.surveys = surveys;
+  this.countTypes = types.count;
+  this.observationalTypes = types.observational;
+  this.interceptTypes = types.intercept;
+  this.freeResponseInterceptTypes = types.freeResponseIntercept;
+  this.freeResponseInterceptBins = bins.freeResponseIntercept;
+}
+
+ChartHelper.prototype.loadChart = function(chartId, chartTitle, dataSourceId) {
+  /**
+   * Load a Highcharts chart.
+   * @param {String} chartId - The ID attribute of the chart container on the page.
+   * @param {String} chartTitle - The title of the chart to display.
+   * @param {String} dataSourceId - The ID of the primary data source to display.
+   */
+  // Check the type of the result to determine what kind of chart to display
+  var resultType;
+  if (this.surveys.length === 0) {
+    throw new Error('This survey has no collected data.');
+  } else {
+    resultType = this.surveys[0].data[dataSourceId].type;
+  }
+
+  var isCount = this.countTypes.indexOf(resultType) > -1;
+  var isObservational = this.observationalTypes.indexOf(resultType) > -1;
+  var isIntercept = this.interceptTypes.indexOf(resultType) > -1;
+  var isFreeResponseIntercept = this.freeResponseInterceptTypes.indexOf(resultType) > -1;
+
+  if (!(isCount || isObservational || isIntercept || isFreeResponseIntercept)) {
+    // The data object needs to be one of the valid types
+    throw new Error('Not a valid chart type: ' + resultType);
+  } else {
+    // Retrieve chart data depending on the data source type
+    var chartData = [];
+    var yAxisLabel = (isCount) ? 'Median response' : '% of responses';
+    if (isCount) {
+      chartData = this._getCountChartData(dataSourceId, chartTitle);
+    } else if (isObservational) {
+      chartData = this._getObservationalChartData(dataSourceId);
+    } else if (isIntercept) {
+      chartData = this._getInterceptChartData(dataSourceId);
+    } else {
+      chartData = this._getFreeResponseInterceptChartData(dataSourceId);
+    }
+  }
+  // Format the chart data for Highcharts display
+  var categories = [];
+  var series = [{
```

Comment:
When I hover over a chart for a float field I see this: 

<img width="392" alt="Screen Shot 2019-05-13 at 4 03 12 PM" src="https://user-images.githubusercontent.com/1094243/57653977-c6111d80-7598-11e9-8a8e-ebeb5731ed08.png">

I think we should hide or change this, as "series" won't mean anything to the user in this context!

Reply:
Very true. It would make a lot of sense to uniquely identify the Survey in some way here. We've talked about how multiple surveys on the same day are possible, so the date isn't necessarily a good idea -- maybe the date with the `stop_time` of the survey? What do you think?

Reply:
This is actually representing 3 surveys, so maybe a date range?